Framework/Geometry/src/Instrument.cpp

```diff
@@ -1338,6 +1345,28 @@ boost::shared_ptr<ParameterMap> Instrument::makeLegacyParameterMap() const {
   return pmap;
 }
 
+/** Parse the instrument tree and create ComponentInfo and DetectorInfo.
+ *
+ * This can be called for the base instrument once it is completely created, in
+ * particular when it is stored in the InstrumentDataService for reusing it
+ * later and avoiding repeated tree walks if several workspaces with the same
+ * instrument are loaded. */
+void Instrument::parseTree() {
```

Comment:
This method seems a bit odd, Not clear what it's doing from the signature. Maybe `constructInternalBeamline` is more descriptive? A minor point.